lib_eio_windows/sched.ml

```diff
@@ -0,0 +1,364 @@
+  | None ->
+    t.active_ops <- t.active_ops + 1;
+    let waiters = get_waiters t fd in
+    let was_empty = Lwt_dllist.is_empty waiters.write in
+    let node = Lwt_dllist.add_l k waiters.write in
+    if was_empty then update t waiters fd;
+    Fiber_context.set_cancel_fn k.fiber (fun ex ->
+        Lwt_dllist.remove node;
+        t.active_ops <- t.active_ops - 1;
+        enqueue_failed_thread t k ex
+      );
+    next t
+
+let get_enqueue t k = function
+  | Ok v -> enqueue_thread t k v
+  | Error ex -> enqueue_failed_thread t k ex
+
+let await_timeout t (k : unit Suspended.t) time =
+  match Fiber_context.get_error k.fiber with
+  | Some e -> Suspended.discontinue k e
+  | None ->
+    let node = Zzz.add t.sleep_q time k in
+    Fiber_context.set_cancel_fn k.fiber (fun ex ->
+        Zzz.remove t.sleep_q node;
+        enqueue_failed_thread t k ex
+      );
+    next t
+
+let with_op t fn x =
+  t.active_ops <- t.active_ops + 1;
+  match fn x with
+  | r ->
+    t.active_ops <- t.active_ops - 1;
+    r
+  | exception ex ->
+    t.active_ops <- t.active_ops - 1;
+    raise ex
+
+[@@@alert "-unstable"]
+
+type _ Effect.t += Enter : (t -> 'a Eio_utils.Suspended.t -> [`Exit_scheduler]) -> 'a Effect.t
+let enter fn = Effect.perform (Enter fn)
+
+let run ~extra_effects t main x =
+  let rec fork ~new_fiber:fiber fn =
+    let open Effect.Deep in
+    Ctf.note_switch (Fiber_context.tid fiber);
+    match_with fn ()
+      { retc = (fun () -> Fiber_context.destroy fiber; next t);
+        exnc = (fun ex ->
+            Fiber_context.destroy fiber;
+            Printexc.raise_with_backtrace ex (Printexc.get_raw_backtrace ())
+          );
+        effc = fun (type a) (e : a Effect.t) ->
+          match e with
+          | Enter fn -> Some (fun k ->
+              match Fiber_context.get_error fiber with
+              | Some e -> discontinue k e
+              | None -> fn t { Suspended.k; fiber }
+            )
+          | Eio.Private.Effects.Get_context -> Some (fun k -> continue k fiber)
+          | Eio.Private.Effects.Suspend f -> Some (fun k ->
+              let k = { Suspended.k; fiber } in
+              let enqueue = get_enqueue t k in
+              f fiber enqueue;
+              next t
+            )
+          | Eio.Private.Effects.Fork (new_fiber, f) -> Some (fun k ->
+              let k = { Suspended.k; fiber } in
+              enqueue_at_head t k;
+              fork ~new_fiber f
+            )
+          | Eio_unix.Private.Await_readable fd -> Some (fun k ->
+              await_readable t { Suspended.k; fiber } fd
+            )
+          | Eio_unix.Private.Await_writable fd -> Some (fun k ->
+              await_writable t { Suspended.k; fiber } fd
+            )
+          | e -> extra_effects.Effect.Deep.effc e
+      }
+  in
+  let result = ref None in
+  let `Exit_scheduler =
+    let new_fiber = Fiber_context.make_root () in
+    fork ~new_fiber (fun () ->
+        result := Some (with_op t main x);
+      )
```

Comment:
Hmm... Should we install the DLA support here?  (That was recently added to other backends.)

Reply:
See here for the [posix backend version](https://github.com/ocaml-multicore/eio/blob/7ea93f2bc3d5ecc6d4dbf8dc5d6e13e5c8831ff4/lib_eio_posix/sched.ml#L352-L360).

Reply:
Yep! Hopefully supported in https://github.com/ocaml-multicore/eio/pull/497/commits/9a9c02ba080d63477b5cdcbad5999928bcc59bb0 with a test based on the README one.